segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -268,14 +268,11 @@ public QueueStats getQueueStatistics() {
     private void processWritesSync() {
         if (getWriteLedger().ledger.isClosed()) {
             // Current ledger is closed. Execute the rollover processor to safely create a new ledger. This will reinvoke
-            // the write processor upon finish, so
+            // the write processor upon finish, so the writes can be reattempted.
             this.rolloverProcessor.runAsync();
-        } else if (!processPendingWrites()) {
+        } else if (!processPendingWrites() && !this.closed.get()) {
```

Comment:
As I understand it, this change is simply making sure that we don't retry if this `BookKeeperLog` object is closed. It sounds like a small bug that is being fixed, but is unrelated to the core issue of this PR, is my understanding correct?